presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -306,6 +307,18 @@ protected Boolean visitInPredicate(InPredicate node, Void context)
         protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
             if (metadata.isAggregationFunction(node.getName())) {
+                if (node.getSortItems().isPresent()) {
+                    for (SortItem sortItem : node.getSortItems().get()) {
+                        Expression sortKey = sortItem.getSortKey();
```

Comment:
please check explicitly that `sortKey` doesn't reference output columns, something similar to this should work:
```
                    // ensure that no output fields are referenced from ORDER BY clause
                    if (orderByScope.isPresent()) {
                        verifyNoOrderByReferencesToOutputColumns(expression);
                    }
```
but with a proper error message.
Add tests for this.